lib/internal/Magento/Framework/Serialize/Serializer/JsonHexTag.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Framework\Serialize\Serializer;
+
+use Magento\Framework\Serialize\SerializerInterface;
+
+/**
+ * Serialize data to JSON, unserialize JSON encoded data
+ *
+ * @api
+ * @since 100.2.0
+ */
+class JsonHexTag extends Json implements SerializerInterface
+{
+    /**
+     * {@inheritDoc}
+     * @since 100.2.0
+     */
+    public function serialize($data): string
```

Comment:
It would be best to keep signature aligned with the interface

Reply:
If i remember well, the tests wanted return type declaration but returning a string|bool type as in the interface isn't yet allowed so i had to return a string (or what do you suggest i return?).

Reply:
The return type added in interface implementors souldn't result in any errors (unlike i.e. adding the parameter type) so I think the implementation is fine